### PR TITLE
treewide: replace `mkPackageOptionMD` with `mkPackageOption`

### PR DESCRIPTION
**Pull request by h7x4**
## Description of changes

I've replaced all occurences of `mkPackageOptionMD` with `mkPacageOption`, and added a deprecation warning to the function.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Kranzes** [comment]
Do they have the same functionality?

**Comment by h7x4** [comment]
Yes, these days one is an alias of the other

**Review by bendlas — APPROVED** [review]
Looks reasonable to me, but I'm not familiar with the context, so please wait for other reviews as well.

**Comment by infinisil** [comment]
Ping @pennae who's the most knowledgeable about the markdown transition :)

**Review by pennae — COMMENTED** [review]
looks good overall. `optionsJSON` output is identical, so we can be sure nothing broke. only thing we disagree with is adding a deprecating `warn` already (and without even mentioning it in the release notes)

File: lib/options.nix [hunk 1]
```diff
@@ -223,7 +223,7 @@ rec {
   /* Alias of mkPackageOption. Previously used to create options with markdown
      documentation, which is no longer required.
   */
-  mkPackageOptionMD = mkPackageOption;
+  mkPackageOptionMD = lib.warn "mkPackageOptionMD is a deprecated alias of mkPackageOption." mkPackageOption;
```
[the remark this answers is not in the document]
Comment (h7x4) [reply lib/options.nix]: That makes sense. I've removed the commit and added a new commit with a deprecation notice in the release notes.

**Review by pennae — APPROVED** [review]

File: lib/options.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (infinisil) [reply lib/options.nix]: I don't think it makes sense to deprecate something without a warning. I'd rather not have anything in the release notes for now.

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 2]
```diff
@@ -31,6 +31,10 @@ In addition to numerous new and upgraded packages, this release has the followin
   The `nimPackages` and `nim2Packages` sets have been removed.
   See https://nixos.org/manual/nixpkgs/unstable#nim for more information.
 
+- `lib.options.mkPackageOptionMD` is now deprecated. It is an alias of
+  `lib.options.mkPackageOption`, and any invocations should be changed
+  to use it's non `MD` equivalent.
+
```
Comment (infinisil) [inline nixos/doc/manual/release-notes/rl-2405.section.md:37]: I don't think it makes sense to deprecate something without a warning. I'd rather not have anything in the release notes for now.

```suggestion
```

**Review by infinisil — CHANGES_REQUESTED** [review]

File: nixos/doc/manual/release-notes/rl-2405.section.md (hunk 2 above)
Comment (h7x4) [reply nixos/doc/manual/release-notes/rl-2405.section.md]: Removed last commit

**Review by infinisil — APPROVED** [review]

**Pull request merged**